redisgears_v8_plugin/src/v8_script_ctx.rs

```diff
@@ -142,6 +169,213 @@ impl V8ScriptCtx {
+        args: Option<&[&V8LocalValue<'isolate_scope, 'isolate>]>,
+        gil_statuc: GilStatus,
+    ) -> Option<V8LocalValue<'isolate_scope, 'isolate>> {
+        let old_val = self.before_run();
+        if gil_statuc.is_locked() {
+            self.after_lock_gil();
+        }
+        let res = func.call(&ctx_scope, args);
+        if gil_statuc.is_locked() {
+            self.before_release_gil();
+        }
+        self.after_run(old_val);
+        res
+    }
+
+    /// The function calls the specified V8 script.
+    /// It performs the necessary operations before and after invoking the function,
+    /// such as setting a variable to indicate that JavaScript code is currently running
+    /// or recording the time at which the GIL (Global Interpreter Lock) is locked for timeout support.
+    pub(crate) fn run<'isolate_scope, 'isolate>(
+        &self,
+        script: &V8LocalScript<'isolate_scope, 'isolate>,
+        ctx_scope: &V8ContextScope<'isolate_scope, 'isolate>,
+        gil_statuc: GilStatus,
+    ) -> Option<V8LocalValue<'isolate_scope, 'isolate>> {
+        let old_val = self.before_run();
+        if gil_statuc.is_locked() {
+            self.after_lock_gil();
+        }
+        let res = script.run(&ctx_scope);
+        if gil_statuc.is_locked() {
+            self.before_release_gil();
+        }
+        self.after_run(old_val);
+        res
+    }
+
+    /// Resolve the given promise object with the given value.
+    /// It performs the necessary operations before and after invoking the function,
+    /// such as setting a variable to indicate that JavaScript code is currently running.
+    pub(crate) fn resolve(
+        &self,
+        resolver: &V8LocalResolver,
+        ctx_scope: &V8ContextScope,
+        val: &V8LocalValue,
+    ) {
+        let old_val = self.before_run();
+        resolver.resolve(ctx_scope, val);
+        self.after_run(old_val);
+    }
+
+    /// Reject the given promise object with the given value.
+    /// It performs the necessary operations before and after invoking the function,
+    /// such as setting a variable to indicate that JavaScript code is currently running.
+    pub(crate) fn reject(
+        &self,
+        resolver: &V8LocalResolver,
+        ctx_scope: &V8ContextScope,
+        val: &V8LocalValue,
+    ) {
+        let old_val = self.before_run();
+        resolver.reject(ctx_scope, val);
+        self.after_run(old_val);
+    }
+
+    /// Runs the given closure if the given promise obejct was already resolved or rejected.
+    fn run_on_done_promise<
+        'isolate_scope,
+        'isolate,
+        'ctx_scope,
+        T,
+        Done: FnOnce(Result<OnDoneCtx, GearsApiError>) -> T,
+    >(
+        &self,
+        isolate_scope: &'isolate_scope V8IsolateScope<'isolate>,
+        ctx_scope: &'ctx_scope V8ContextScope<'isolate_scope, 'isolate>,
+        promise: &V8LocalPromise<'isolate_scope, 'isolate>,
+        on_done: Done,
+    ) -> T {
```

Comment:
I am not sure it will look better, for sure it will not change the calling of the functions and we will still need all the generic arguments. Am I missing something?